### ISSUE TITLE
XSOverlay won't launch if AllowAdminPermissions is set to true

### ISSUE THREAD

**Issue by tasonosenshi** [Xiexe/XSOverlay-Issue-Tracker#322]
**Describe the bug in detail:**
On a fresh install of XSOverlay, I checked the setting to make it launch with Admin permissions so I could interact with task manager while in VR.  After closing the application using Steam, I could no longer re-launch the application.  Both auto-launch on SteamVR start and directly launching in Steam fail to start the application.  I've tested both regular and beta options in steam.

**Provide steps to reproduce the bug:**
  1. Uninstall XSOverlay
  2. Manually delete `C:\Users\<user>\AppData\LocalLow\Xiexe\XSOverlay`
  3. Reinstall XSOverlay
  4. Launch XSOverlay
  5. Open settings, and check the box to launch with Admin permissions
  6. Use Steam to stop XSOverlay
  7. XSOverlay won't launch again

**Expected behavior:**
XSOverlay launches, maybe with a UAC prompt the first time, and then I can spawn an overlay that can interact with task manager.

**Output Log:**
You can find your output log at `C:\Users\[Username]\AppData\LocalLow\Xiexe\XSOverlay\`
You can find the uWindowCapture and uDesktopCapture log in the XSOverlay install directory.
Please attach your logs to the issue.

No logs exist in either location after this process.  It doesn't seem to get far enough through startup to log anything?

**Additional Information:**
OS Name	Microsoft Windows 11 Pro
Version	10.0.22631 Build 22631

From the Steam Properties menu, Updates tab:
Regular XSOverlay Build ID: 12586499
Beta XSOverlay Build ID: 13427328

**Comment by tasonosenshi** [comment]
Update:  

Deleting `C:\Users\<user>\AppData\LocalLow\Xiexe\XSOverlay\config\overlaySettings.json` allows XSOverlay to launch again.

Also, instead of deleting the file, simply editing the field `AllowAdminPermissions` to `false` seems to allow XSOverlay to launch as well,

**Comment by Riletin** [comment]
Also having this issue on Windows 10 Home.

**Comment by Xiexe** [comment]
Hi all!

Update here, I've found a solution to this and the root cause of the issue seems to be that the file path has changed.

That could be either due to the user moving the install directory or due to a windows reinstall, or even just installing a new hard drive causing the drive letter to change.

There will be a fix out this week for this, and ill update the issue here when I push the update!

**Comment by tasonosenshi** [comment]
I'm still seeing this issue on 
Build ID: 14028712

If the issues is a different filepath, that makes sense.  My Steam library that XSOverlay is in is under `C://Steam`.

I had to move my steam folder on my main drive to accommodate some extensive mods on an unrelated game.

**Comment by hazre** [comment]
Ok none of the workarounds actually work, so I investigated it myself.

The actual problem is due to XSOverlay not properly cleaning up Task Scheduler with their Tasks pointing to completely wrong paths. (for e.g. it was pointed to `D:\SteamLibrary\steamapps\common\XSOverlay_Beta\Launcher/../XSOverlay.exe` for me, when it should've been `C:\Program Files (x86)\Steam\steamapps\common\XSOverlay_Beta\XSOverlay.exe`

Here's how you fix it and it should let you use XSOverlay with Admin settings enabled:

1. Enable AllowAdminPermissions either via overlay itself or by editing `C:\Users\%username%\AppData\LocalLow\Xiexe\XSOverlay\config\overlaySettings.json` directly.
2. Open Task Scheduler, under Task Scheduler Library, Look for `XSOverlay Launcher` and `XSOverlay Launcher Supporter` and delete both.
3. (ONE TIME ONLY) Navigate to where XSOverlay is installed, there should be a folder called `Launcher`, open that and then there should be a executable there called `XSOverlay Process Manager.exe`. Right click on that and run as Administrator. (full path for me is this: `C:\Program Files (x86)\Steam\steamapps\common\XSOverlay_Beta\Launcher\XSOverlay Process Manager.exe`)
4. You are pretty much done but to make sure it works now, Kill all XSOverlay processes (`XSOverlay.exe`, `XSOverlay Process Manager.exe`, `XSOverlay Media Manager.exe`) from Task Manager. Now try opening XSOverlay normally through Steam (Yes that also means through SteamVR startup apps) and now you should be able to open, move and etc privileged windows such as Task Manager.

TL;DR this recreates the Tasks in Task Scheduler, so XSOverlay can open with admin perms through those tasks without any UAC prompts.

**Comment by staringrail** [comment]
@hazre This worked flawlessly. Thank you so much for this guide.

**Comment by FluffyPuppyKasey** [comment]
This issue still persists. It's been nearly a year. WHY isn't this fixed yet? Your average user isn't going to dig through task scheduler to find stuff, they're just gonna uninstall and get a refund. This is a FUNDAMENTAL issue and honestly XSOverlay shouldn't be paid if @Xiexe isn't going to fix this

**Comment by Xiexe** [comment]
> This issue still persists. It's been nearly a year. WHY isn't this fixed yet? Your average user isn't going to dig through task scheduler to find stuff, they're just gonna uninstall and get a refund. This is a FUNDAMENTAL issue and honestly XSOverlay shouldn't be paid if @Xiexe isn't going to fix this

This has been fixed on public beta for about a week and a half, just waiting to hear feedback on whether or not the beta causes any other stability issues.

Please refer to the steam changelog before flaming. Thanks.

**Comment by FluffyPuppyKasey** [comment]
> > This issue still persists. It's been nearly a year. WHY isn't this fixed yet? Your average user isn't going to dig through task scheduler to find stuff, they're just gonna uninstall and get a refund. This is a FUNDAMENTAL issue and honestly XSOverlay shouldn't be paid if [@Xiexe](https://github.com/Xiexe) isn't going to fix this
> 
> This has been fixed on public beta for about a week and a half, just waiting to hear feedback on whether or not the beta causes any other stability issues.
> 
> Please refer to the steam changelog before flaming. Thanks.

While I do get needing to test, honestly a change that fixes a flaw that severe should already been pushed to the public branch, even if that's the only change, considering it's been a year since the issue was first reported

**Comment by Xiexe** [comment]
> > > This issue still persists. It's been nearly a year. WHY isn't this fixed yet? Your average user isn't going to dig through task scheduler to find stuff, they're just gonna uninstall and get a refund. This is a FUNDAMENTAL issue and honestly XSOverlay shouldn't be paid if [@Xiexe](https://github.com/Xiexe) isn't going to fix this
> > 
> > This has been fixed on public beta for about a week and a half, just waiting to hear feedback on whether or not the beta causes any other stability issues.
> > 
> > Please refer to the steam changelog before flaming. Thanks.
> 
> While I do get needing to test, honestly a change that fixes a flaw that severe should already been pushed to the public branch, even if that's the only change, considering it's been a year since the issue was first reported

While i appreciate that it's a large issue for you personally, I have analytics that tell me that a majority of users do not use admin mode. While power users might, average users don't. Much less, changing the install directory, though I don't have data on that- I can't imagine it's very common, which is why it hasn't been a priority.

As far as pushing fixes quickly / slowly, once you've published a piece of software your thought process can change on things like that. I used to push updates quick and dirty but quickly realized after the user base grew that I couldn't do that without risk of severe backlash.

Anyways, this isn't the place for conversations like this - the point is that it should be fixed, and you can access it. You'll just need to wait for a bit before it gets merged to the main branch.
